File: validator/python/src/generate_data_questions.py
from src.question import Question
from src.test import Test

def generate_data_questions(json_data_questions):
  questions = []

  for question in json_data_questions:
    tests = [Test(test["args"], test["result"]) for test in question["tests"]]
    questions.append(Question(question["name"], tests))

  return questions

File: validator/javascript/src/generate-data-questions.js
const { Question } = require("./question");
const { Test } = require("./test");

function generateDataQuestions(jsonDataQuestions) {
  const questions = {}

  for (const [keyQuestion, question] of Object.entries(jsonDataQuestions)) {
    const tests = question.tests.map(({ args, result }) => new Test(args, result));
    questions[keyQuestion] = new Question(keyQuestion, tests);
  }

  return questions
}

module.exports = {
  generateDataQuestions
}